proc_test.go

```diff
@@ -175,3 +178,19 @@ func TestHelloWorld(t *testing.T) {
 	)
 	proc.Run(t)
 }
+
+func TestFrameCount(t *testing.T) {
+	proc := newTestProc(t, defaultWidth, defaultHeight,
+		func(proc *Proc) {},
+		func(proc *Proc) {},
+		"",
+	)
+	if fc := proc.FrameCount(); fc != 0 {
+		t.Errorf("initial frame count should be 0, got %d", fc)
+	}
+
+	proc.Run(t)
+	if fc := proc.FrameCount(); fc != 1 {
```

Comment:
could you also add a test with "no-loop" to check the frame-count stays at 0?

Reply:
Can you please give me an example how to do this?

- Calling testProc.Run call directly draw() once without any checks if loop is enabled
- Calling Proc.Run inside a test does not work because "Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: '+[NSUndoManager(NSInternal) _endTopLevelGroupings] is only safe to invoke on the main thread."
- Calling Proc.run goes into a loop so there is no way to actually stop it. A system.DestroyEvent could do the job but the app.Window instance is created inside the run method, so there is no way to send a specific event. Making app.Window part of the Proc struct could help a lot with testing if custom events can be send to app.Window.in channel.

Reply:
ah, right, one would need to add

Reply:
the with-loop/no-loop tests could be split into 2 different `TestFrameCount` functions, or within the same test func but 2 different `testProc` values.

Reply:
But adding a 

`if p.Proc.loop() {`

around draw inside the proc_test, means that we test against the test code and not the "production" code.

What needs to be tested is p5.Proc.run() https://github.com/go-p5/p5/blob/main/proc.go#L173 which already includes a loop check here https://github.com/go-p5/p5/blob/main/proc.go#L239, correct?

Do I miss something fundamental here?